client/src/utils/encodeForm.ts

```diff
@@ -0,0 +1,8 @@
+/**
+ * Encodes form data into a URL-encoded string (application/x-www-form-urlencoded).
```

Comment:
Review the issue, there's an example implementation that's closer aligned to the way jQuery handles this. 

Forms can support files, this code needs to handle that gracefully. 

Also. We will need unit tests for this new utility.